### PR TITLE
414 fulfilling posts

### LINKED ISSUE #414: Track whether requests have been fulfilled
- [X] Add a way for post creators to mark their request or offer as fulfilled.
- [X] Style fulfilled requests/offers differently

Depends on: https://github.com/Hylozoic/hylo-node/issues/464

### PR DESCRIPTION
**Pull request by cpolitano**
[Closes #414 ]

When a post is viewed in expanded view by its creator, a dropdown appears to let the creator select whether the request/offer is still needed/available or was fulfilled. Can be reversed.

Offers & Requests have optional timeframes when they are created, which display in the post headers.

When a post is marked Completed by the creator, a Completed badge appears in the post header.

Screen Shot 2020-05-01 at 4 50 02 PM

Screen Shot 2020-05-12 at 10 07 25 AM

Screen Shot 2020-05-12 at 10 29 12 AM

### CODE REVIEW

File: src/components/PostCard/PostTitle/PostTitle.js [hunk 1]
```diff
@@ -1,13 +1,11 @@
 import React from 'react'
-import { decode } from 'ent'
 import Highlight from 'components/Highlight'
 import './PostTitle.scss'
 
 export default function PostTitle ({
   title,
   highlightProps
 }) {
-  title = decode(title)
```
Comment (tibetsprague) [inline src/components/PostCard/PostTitle/PostTitle.js:10]: why remove this?

File: src/components/PostCard/PostHeader/PostHeader.js [hunk 2]
```diff
@@ -13,6 +14,27 @@ import './PostHeader.scss'
 import { filter, isFunction, isEmpty } from 'lodash'
 import cx from 'classnames'
 
+const getStartDate = (startTime) => {
```
Comment (tibetsprague) [inline src/components/PostCard/PostHeader/PostHeader.js:17]: i might call this formatStartDate

File: src/components/PostCard/PostTitle/PostTitle.js (hunk 1 above)
Comment (cpolitano) [reply src/components/PostCard/PostTitle/PostTitle.js:10]: I changed how that prop is pass - it's passed as a string now, not an HTML entity and no longer needs to be decoded.

File: src/components/PostCard/PostHeader/PostHeader.js (hunk 2 above)
Comment (cpolitano) [reply src/components/PostCard/PostHeader/PostHeader.js:17]: will update

File: src/components/PostCard/PostCompletion/PostCompletion.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import React from 'react'
+import './PostCompletion.scss'
+import DropdownButton from 'components/DropdownButton'
+
+export default function PostCompletion ({ type, startTime, endTime, isFulfilled, fulfillPost, unfulfillPost }) {
+  var label
+
+  switch (isFulfilled) {
+    case false:
+      type === 'request' ? label = 'I still need this' : label = 'Available'
```
[the remark this answers is not in the document]
Comment (cpolitano) [reply src/components/PostCard/PostCompletion/PostCompletion.js:10]: ok thanks

**Pull request merged**